bentoml/yatai/deployment/azure_function/operator.py

```diff
@@ -0,0 +1,724 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+import re
+import shutil
+import subprocess
+import logging
+import sys
+
+import docker
+
+from bentoml import config
+from bentoml.utils.tempdir import TempDirectory
+from bentoml.saved_bundle import loader
+from bentoml.yatai.deployment.azure_function.templates import (
+    AZURE_FUNCTION_HOST_JSON,
+    AZURE_FUNCTION_LOCAL_SETTING_JSON,
+    AZURE_API_INIT_PY,
+    AZURE_API_FUNCTION_JSON,
+    BENTO_SERVICE_AZURE_FUNCTION_DOCKERFILE,
+)
+from bentoml.yatai.deployment.operator import DeploymentOperatorBase
+from bentoml.yatai.deployment.utils import ensure_docker_available_or_raise
+from bentoml.exceptions import (
+    BentoMLException,
+    MissingDependencyException,
+    AzureServiceError,
+    YataiDeploymentException,
+)
+from bentoml.yatai.proto.deployment_pb2 import (
+    ApplyDeploymentResponse,
+    DeploymentState,
+    DescribeDeploymentResponse,
+    DeleteDeploymentResponse,
+)
+from bentoml.yatai.proto.repository_pb2 import GetBentoRequest
+from bentoml.yatai.status import Status
+from bentoml.configuration import LAST_PYPI_RELEASE_VERSION
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_azure_cli_available_and_login_or_raise():
+    try:
+        account_info = _call_az_cli(
+            command=['az', 'account', 'show'], message='show Azure account'
+        )
+        if not account_info['user'] or not account_info['homeTenantId']:
+            # https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest
```

Comment:
Looking at this doc, I feel it's not a good idea for BentoML to manage the azure CLI login here - there are quite a few ways a user can log in and manage azure permission, and here we lose support for things like parameters like VaultName and CertName.

It seems better to just let the user use environment variable or mounting ` ~\.azure\accessTokens.json` file for azure login